dask/base.py

```diff
@@ -1350,29 +1362,34 @@ def normalize_datatype(dt):
 def register_numpy():
     import numpy as np
 
+    @normalize_token.register(np.ndarray)
+    def normalize_array(x):
+        if not x.shape:
+            return (x.item(), x.dtype)
+        if x.dtype.hasobject:
+            try:
+                try:
+                    # string fast-path
+                    data = hash_buffer_hex(
+                        "-".join(x.flat).encode(
+                            encoding="utf-8", errors="surrogatepass"
+                        )
+                    )
+                except UnicodeDecodeError:
+                    # bytes fast-path
+                    data = hash_buffer_hex(b"-".join(x.flat))
+            except (TypeError, UnicodeDecodeError):
+                return normalize_object(x)
+        else:
+            try:
+                data = hash_buffer_hex(x.ravel(order="K").view("i1"))
+            except (BufferError, AttributeError, ValueError):
+                data = hash_buffer_hex(x.copy().ravel(order="K").view("i1"))
+        return (data, x.dtype, x.shape)
+
     @normalize_token.register(np.memmap)
     def normalize_mmap(mm):
-        if hasattr(mm, "mode") and getattr(mm, "filename", None):
```

Comment:
Why are we removing this?

Reply:
dead code

Reply:
oh hold on, I think this is a mistake. one sec

Reply:
yeah, so pickle/cloudpickle also does funky stuff to mmaps such that this tokenization is not stable. 
I replaced the normalize_obejct with a hash buffer